modules/peers.js

```diff
@@ -439,13 +442,14 @@ Peers.prototype.discover = function (cb) {
  * @return {peer[]} Filtered list of peers
  */
 Peers.prototype.acceptable = function (peers) {
-	return _.chain(peers).filter(function (peer) {
-		// Removing peers with private or address or with the same nonce
-		return !ip.isPrivate(peer.ip) && peer.nonce !== library.nonce;
-	}).uniqWith(function (a, b) {
-		// Removing non-unique peers
-		return (a.ip + a.port) === (b.ip + b.port);
-	}).value();
+	return _(peers)
+		.uniqWith(function (a, b) {
+			// Removing non-unique peers
+			return (a.ip + a.port) === (b.ip + b.port);
+		})
+		.filter(function (peer) {
+			return !ip.isPrivate(peer.ip) && peer.nonce !== modules.system.getNonce();
```

Comment:
Please add comment: `// Removing peers with private address or nonce equal to self`